server/src/api/v1/Invite/index.ts

```diff
@@ -0,0 +1,85 @@
+import moment from 'moment';
+import Router from '../../../lib/router';
+import sendMail from '../../../lib/mail';
+import emailTemplate from '../../../lib/mail/emailTemplate';
+import logger from '../../../logger';
+import { Invite } from '../../../models';
+import { DEFAULT_INVITE_EXPIRE_TIME_HOURS } from '../../../models/Invite/index.model';
+import UserService from '../../../lib/user';
+
+const router = Router();
+
+router.post('/:inviteId/verify', async (req, res) => {
+  const { inviteId } = req.params;
+  let invite: Invite = null;
+  try {
+    invite = await Invite.findOne({
+      where: {
+        id: inviteId,
+      },
+    });
+  } catch (e) {
+    res.json({
+      status: 'ERROR',
+      msg: 'invite not found',
+    });
+    return;
+  }
+
+  if (!invite) {
+    res.json({
+      status: 'ERROR',
+      msg: 'invite not found',
+    });
+    return;
+  }
+
+  const { userId } = invite;
+  const user = await UserService.find(userId);
+  const operationSettings: any = {}; // user.operationId to get settings
+  const now: Date = new Date();
+
+  const expireTime = operationSettings.inviteExpireTime || DEFAULT_INVITE_EXPIRE_TIME_HOURS;
+  if (moment(invite.sentAt).add(expireTime, 'hours').isAfter(now)) {
+    invite.update({ approvedAt: now });
+    user.update({ emailVerified: true });
```

Comment:
i would put line 40-45 in another function in lib/invite.ts and check if it has result or null for returning an error